ca/client.go

```diff
@@ -97,12 +98,43 @@ func (c *uaClient) PostWithContext(ctx context.Context, u, contentType string, b
 		return nil, errors.Wrapf(err, "create POST %s request failed", u)
 	}
 	req.Header.Set("Content-Type", contentType)
-	req.Header.Set("User-Agent", UserAgent)
-	return c.Client.Do(req)
+	return c.Do(req)
+}
+
+// requestIDHeader is the header name used for propagating request IDs from
+// the CA client to the CA and back again.
+const requestIDHeader = "X-Request-Id"
+
+// newRequestID generates a new random UUIDv4 request ID. If it fails,
+// the request ID will be the empty string.
+func newRequestID() string {
+	requestID, err := randutil.UUIDv4()
+	if err != nil {
+		return ""
```

Comment:
Is it possible for `randutil.UUIDv4` to fail? If yes, then there's something wrong with the entropy of the underlying system and that's something we should take note of.

Reply:
It is possible to fail, which is why an error is returned. It can fail on a client system, but there's not much we can do, unless we want an alternative random ID to be generated that doesn't rely on the failing code path. 

Given that the server side will generate a request ID if the client doesn't provide one, IMO we don't need that additional method for generating a unique ID.